### PR TITLE
Update fields.md

### PR DESCRIPTION
**Pull request by etrotta**
Fix a code block under the "VSCode and Pyright users" section
Trivial change to the documentation

Selected Reviewer: @Kludex

### CODE REVIEW

File: docs/usage/fields.md [hunk 1]
```diff
@@ -191,7 +191,7 @@ print(user.model_dump(by_alias=True))  # (2)!
     class User(BaseModel):
         model_config = ConfigDict(populate_by_name=True)
 
-        name: str = Field(..., alias='username')
+        name: str = Field(..., alias=str('username'))
```
Comment (hramezani) [inline docs/usage/fields.md:194]: we don't need this change.

What is the reason for this change here?

Comment (etrotta) [reply docs/usage/fields.md:194]: If that is not required, then this part of the same document is wrong then?
> To "trick" VSCode into preferring the field name, you can use the `str` function to wrap the alias value

I found it weird that there are two code blocks essentially equal to each other and the code does not matches the description above it

**Comment by hramezani** [comment]
please review

**Comment by Kludex** [comment]
What's the issue for this PR?

**Comment by etrotta** [comment]
> @etrotta What's the issue for this PR?
> 
> I do remember something about this...

I was hoping it would be just a trivial change to the documentation so did not open one, but from that section of the documentation it mentions https://github.com/pydantic/pydantic/issues/5893

**Comment by samuelcolvin** [comment]
this conflicts with main since #7322 was merged. please update.
